sdk/identity/Azure.Identity/tests/TokenCredentialOptionsTests.cs

```diff
@@ -101,5 +110,171 @@ public void EnviornmentSpecifiedNonHttpsAuthorityHostFails()
                 Assert.Throws<ArgumentException>(() => new UsernamePasswordCredential(tenantId, clientId, username, password, new TokenCredentialOptions()));
             }
         }
+
+        [Test]
+        public void VerifyClone([Values]SupportsInterfaces originalSupportsInterfaces, [Values]SupportsInterfaces cloneSupportsInterfaces, [Values]bool setTransport)
```

Comment:
I think we should consider implementing this test using all the real credential types and using reflection to evaluate all implemented interfaces. The reason being that, as is, this test would continue to pass if we were to add a new `ISupports*` interface and forget to add the logic for it to the Clone method.